src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -457,7 +458,7 @@ public SslMode SslMode
         [Description("Whether to trust the server certificate without validating it.")]
         [DisplayName("Trust Server Certificate")]
         [NpgsqlConnectionStringProperty]
-        public bool TrustServerCertificate
+        public bool? TrustServerCertificate
```

Comment:
Based on our last decision, I don't think this needs to be nullable... This can remain false by default, and we simply throw for Require if it's not true.

/cc @NinoFloris